geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -1550,12 +1574,64 @@ public void refuseHandshake(OutputStream out, String message, byte exception) th
     out.flush();
   }
 
-  private boolean handOffQueueInitialization(Socket socket, CommunicationMode communicationMode) {
+  @Override
+  public void refuseHandshake(OutputStream out, String message, byte exception,
+      NioSslEngine sslEngine, Socket socket) throws IOException {
+    if (sslEngine == null) {
+      refuseHandshake(out, message, exception);
+      return;
+    }
+    try (ByteBufferOutputStream bbos =
+        new ByteBufferOutputStream(sslEngine.getEngine().getSession().getPacketBufferSize())) {
```

Comment:
Here's another example of breaking the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) and exposing internal knowledge that the acceptor shouldn't have to know about. It's better to embrace dependency injection and just pass in `int packetBufferSize` or even `javax.net.ssl.SSLSession sslSession` as a parameter.

One direction I think we really need to steer the code base towards is [dependency injection](https://en.wikipedia.org/wiki/Dependency_inversion_principle). The idea is to pass in all dependencies, either to the constructor or as parameters to a method call, instead of reaching out to fetch them.

Reply:
Thanks for comments. Updated.